lib/plugins/plugin_api.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Plugin Class Extension for API Type Plugins
+class Plugin
+  private
+
+  def exec_api(data_from_chat, chat_text_array)
+    response = api_response(data_from_chat, chat_text_array)
+
+    if response.success?
+      response.body unless response.body.blank?
+      response.code if @settings.environment == 'test'
+    else
+      @logger.error("Plugin: #{@name}: returned code of #{response.code}")
+      return "Error: Received code #{response.code}"
+    end
+  end
+
+  def api_response(data_from_chat, chat_text_array)
+    payload = api_payload(data_from_chat, chat_text_array)
+    exec_url = "#{@api_url}#{@config['api_config']['endpoint']}"
+    @logger.debug("Plugin: #{@name}: Exec URL is set to #{exec_url}")
+    @logger.debug("Plugin: #{@name}: Exec '#{chat_text_array.drop(2)}' being sent via API")
+    auth = @config.dig('plugin', 'api_config', 'basic_auth') ? @config['api_config']['basic_auth'] : false
+    api_call(payload, auth, exec_url)
+  end
+
+  def api_call(payload, auth, exec_url)
+    retry_count = 0
+    while retry_count < 5
+      begin
+        response = HTTParty.post(exec_url, basic_auth: auth, body: payload, headers: @headers) if auth
+        response = HTTParty.post(exec_url, body: payload, headers: @headers) unless auth
+        break unless response.code.blank?
+        retry_count += 1
+      rescue Errno::ECONNREFUSED
+        api_timeout_catch(retry_count)
+        retry_count += 1
+      end
+    end
+    response
+  end
+
+  def api_timeout_catch(count)
+    raise "[ERROR] Plugin: #{@name}: Too many retries, plugin is not reachable" if count >= 5
+    @logger.info("Plugin: #{@name}: connection refused; retrying in 5s...") unless count >= 5
```

Comment:
They are opposites based on the if/unless, however 5 is part of both.

`if count >= 5` means 5, 6, 7, 8 ....
`unless count >= 5` means 0, 1, 2, 3, 4, 5

likely it should be if in both cases for readability.  However if you change the second one it should be:

``if count < 5` note that there is not an "=" in that.